### PR TITLE
ci: restructured, cleaned, added pub.dev checks

### PR DESCRIPTION
**Pull request by daadu**
- [deploy_demo]: proper step name, switched to "stable"
- [pr_title]: proper step name

### CODE REVIEW

**Comment by daadu** [comment]
@WieFel Will release v0.7.0 after this merge.

File: .github/workflows/analysis.yml
Comment (WieFel) [inline .github/workflows/analysis.yml:5]: I would maybe name this `analysis`, just like the filename!?

Comment (daadu) [reply .github/workflows/analysis.yml]: Done more changes - changed file names completely - so that can have more than 1 job in a file - file/action name should suggest - when this should happen - instead of what (job name should be based on what).

**Comment by daadu** [comment]
@WieFel Can you review now.

**Comment by WieFel** [comment]
@daadu what I find a bit strange is using `.sh` files for installing flutter + dependencies. Why did you come up with that solution, instead of using existing github actions that can be included?

**Comment by daadu** [comment]
> @daadu what I find a bit strange is using `.sh` files for installing flutter + dependencies. Why did you come up with that solution, instead of using existing github actions that can be included?

Mainly to make the step faster - check [new - 59s](https://github.com/fluttercommunity/backdrop/runs/4573891405?check_suite_focus=true) vs [old - 66s](https://github.com/fluttercommunity/backdrop/runs/4573894959?check_suite_focus=true), although the difference is not much, I expected more though!

**Comment by daadu** [comment]
I was expecting the 3rd party action to be using caching - for flutter binary - but looks like they don't. May be with our script can add caching to make it faster.

**Comment by WieFel** [comment]
> > @daadu what I find a bit strange is using `.sh` files for installing flutter + dependencies. Why did you come up with that solution, instead of using existing github actions that can be included?
> 
> Mainly to make the step faster - check [new - 59s](https://github.com/fluttercommunity/backdrop/runs/4573891405?check_suite_focus=true) vs [old - 66s](https://github.com/fluttercommunity/backdrop/runs/4573894959?check_suite_focus=true), although the difference is not much, I expected more though!

In my opinion, readability/clarity of the actions would be more important in this case than 5s of quicker execution.
But of course, if you prefer it this way it is also fine with me ;) you have the final word.

**Comment by daadu** [comment]
> In my opinion, readability/clarity of the actions would be more important in this case than 5s of quicker execution. But of course, if you prefer it this way it is also fine with me ;) you have the final word.

going with your suggestion. makes sense.

**Review by WieFel — APPROVED** [review]

**Comment by daadu** [comment]
@WieFel The total difference on run is of 30s - [current](https://github.com/fluttercommunity/backdrop/actions/runs/1598317861) vs [previous](https://github.com/fluttercommunity/backdrop/actions/runs/1598284746)

**Comment by daadu** [comment]
We can try adding caching to current to reduce time or come back to `git clone` approach. Anyways not important for now, just for note - if in future the CI becomes slower - we might have to think on it.

**Comment by WieFel** [comment]
@daadu  ok!
Yeah I think 30s difference is ok for now. Github actions are actually not trimmed for running very fast anyway. Their purpose is to run on some events and give feedback, regardless needing few seconds more or less 😉

**Comment by daadu** [comment]
I think CI feedback should be fast though - like in about a 1-2m should now things are ok!

**Comment by WieFel** [comment]
Yeah sure! As long as it is within this limit it is ok

**Pull request merged**
